src/main/java/org/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -383,27 +386,38 @@ private void connect(boolean autoDetect) {
             // Show progress dialog:
             progressDialog = new DetectOpenOfficeInstallation(diag, preferences, dialogService)
                     .showProgressDialog(diag, Localization.lang("Connecting"), Localization.lang("Please wait..."));
-            getWorker().run(); // Do the actual connection, using Spin to get off the EDT.
-            progressDialog.dispose();
+            JDialog finalProgressDialog = progressDialog;
+            BackgroundTask
+                    .wrap(this::createBibBase)
+                    .onSuccess(ooBase -> {
+                        this.ooBase = ooBase;
+                        SwingUtilities.invokeLater(() -> {
+                            finalProgressDialog.dispose();
+                            diag.dispose();
+                        });
+                        if (ooBase == null) {
+                            //throw connectException; // TODO
```

Comment:
What is the TODO at this place?

Reply:
Good catch, this was a left-over from the error handling before. The error case is now handled better.